docs/design/design_v1.9.md

```diff
@@ -69,196 +67,246 @@ This means we aim to standardize:
  - `--cluster-signing-cert-file` to `ca.crt`, if External CA mode is disabled, otherwise to `""`.
  - `--cluster-signing-key-file` to `ca.key`, if External CA mode is disabled, otherwise to `""`. 
- - `--service-account-private-key-file` to `sa.key`
- - `--use-service-account-credentials` to `true`
+ - `--service-account-private-key-file` to `sa.key`.
+ - `--use-service-account-credentials` to `true`.
 
 #### Scheduler
 
-kubeadm doesn't set any special scheduler flags.
+Kubeadm doesn't set any special scheduler flags.
+
+### Generate static Pod manifest for local etcd
+
+If the user specified an external etcd this step will be skipped, otherwise a static manifest file will be generated for creating a local etcd instance running in a Pod with following attributes:
+
+- listen on `localhost:2379` and use `HostNetwork=true`.
+- make a `hostPath` mount out from the `dataDir` to the host's filesystem.
+- Any extra flags specified by the user.
+
+Please note that:
+
+1. The etcd image will be pulled from `gcr.io/google_containers`. In case an alternative image repository is specified this one will be used; In case an alternative image name is specified, this one will be used.
+2. in case of `kubeadm`  executed in the `--dry-run` mode, the etcd static Pod manifest is written in a temporary folder.
+3. Static Pod manifest generation for local etcd can be invoked individually with the `kubeadm alpha phase etcd local` command.
 
-### Generate Static Pod Manifest for local etcd
+### (optional) Write init kubelet configuration
 
-If the user specified an external etcd this step will be skipped, otherwise a static manifest file will be generated for creating a local etcd instance running in a pod with following attributes:
+If kubeadm is invoked with `--feature-gates=DynamicKubeletConfig`, it writes the kubelet init configuration into `/var/lib/kubelet/config/init/kubelet` file.
 
-- listen on `localhost:2379` and use `HostNetwork=true`
-- make a `hostPath` mount out from the `dataDir` to the host's filesystem
-- Any extra flags specified by the user
+The init configuration is used for starting the kubelet on this specific node, providing an alternative for the kubelet drop-in file; such configuration will be replaced by the kubelet base configuration as described in following steps.
 
 Please note that:
 
-1. The image, for version  `3.0.17` /current architecture, will be pulled from `gcr.io/google_containers`. In case an alternative image repository is specified this one will be used; In case an alternative image name is specified, this one will be used.
-2. in case of `kubeadm`  executed in the `--dry-run` mode, the etcd static pod manifest is written in a temporary folder.
-3. Static Pod Manifest generation for local etcd can be invoked individually with the `kubeadm phase etcd local` command.
+1. to make dynamic kubelet configuration work, flag `--dynamic-config-dir` should be specified in `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf`.
```

Comment:
`--dynamic-config-dir=/var/lib/kubelet/config/dynamic`

and the kubelet with the `--init-config-dir=/var/lib/kubelet/config/init` arg